core/src/com/unciv/logic/civilization/Civilization.kt

```diff
@@ -1083,10 +1084,23 @@ class Civilization : IsPartOfGameInfoSerialization {
         moveCapitalTo(newCapital, oldCapital)
     }
 
-    @Readonly fun getAllyCiv(): Civilization? = if (allyCivName == null) null
-        else gameInfo.getCivilization(allyCivName!!)
-    @Readonly fun getAllyCivName() = allyCivName
-    fun setAllyCiv(newAllyName: String?) { allyCivName = newAllyName }
+    /**
+     * Current ally (assuming this is a city-state)
+     * 
+     * Setting this also sets its backing serialization field ``allyCivName``
+     * */
+    @get:Readonly
+    @Transient
+    var allyCiv: Civilization? = null
```

Comment:
Here, too, I'd prefer this closer to its "serialization" field - and I also prefer not to mix fields and functions in the top-down order of a class.